### DISCUSSION TITLE
Using distrobox-assemble with Fedora COPR and random commands

### DISCUSSION THREAD

**Discussion by boredsquirrel** [89luca89/distrobox#1476] [General]
https://universal-blue.discourse.group/t/2523

Distrobox is not the originally intended use case for Podman containers (as all "state" i.e. things a user wants is done manually and not on the image side), and the `dnf system-upgrade` is broken as it requires a reboot.

Unless uBlue images swap out the packages upstream, users need to recreate the containers with version upgrades.

This is worlds more complicated than the regular "system that just works" with "apps that just work" and never break.

So for a possible GUI app manager or something, `distrobox-assemble` needs to be used.

---

I use a Distrobox for QGis and RStudio, both graphical applications that don't work as Flatpaks currently.

RStudio basically only works on Fedora too, because of the COPR repo including all the plugins you may need. Using a rolling Container that doesnt have this problem is not possible because
- it does not have the packages
- it puts trust in a random other distribution

These are the commands I need to run in a box. I saw no [documentation](https://github.com/89luca89/distrobox/blob/63496a817f65829a5cd457a8ef5d8859d4b9ba9d/docs/usage/distrobox-assemble.md) about adding repos, or if just running commands would be supported.

```
sudo echo "install_weak_deps=False" >> /etc/dnf/dnf.conf
sudo dnf copr enable -y iucar/cran
sudo dnf install -y fish qgis python3-qgis python3-pip rstudio-desktop R-CoprManager
distrobox-export --app qgis
distrobox-export --app rstudio
```

[Distrobox-assemble documentation](https://github.com/89luca89/distrobox/blob/63496a817f65829a5cd457a8ef5d8859d4b9ba9d/docs/usage/distrobox-assemble.md)

**Comment by net-fall** [comment]
For additional repository, you can use `init_hooks`. Something like that:

```ini
init_hooks=dnf copr enable -y lihaohong/yazi
init_hooks=dnf install -y yazi
```
